torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -161,6 +161,43 @@ static inline at::optional<int64_t> parse_as_integer(const std::string& s) {
   return (*str_end == 0) ? at::optional<int64_t>(ans) : at::nullopt;
 }
 
+/*
+Parse default value of IntList declared at native_functions.yaml
+
+There are two kinds of default values:
+1. IntList[2] x=1 (where size=2, value={1,1}
+2. IntList x={1,2,3} (where size=3, value={1,2,3}, note that there cannot be space after comma since native_parse.py uses ', ' to split args)
+*/
+static inline std::vector<int64_t> parse_intlist_args(const std::string& s, int64_t size) {
+  if (s[0] != '{') {
+    return std::vector<int64_t>(size, std::stoi(s));
+  }
+
+  size_t n = s.size();
+  // since already checked left brace '{' above, here only checks right brace '}'
+  AT_CHECK(s[n - 1] == '}', "Default value of IntList is missing right brace '}', found ", s[n - 1]);
+
+  auto args = std::vector<int64_t>();
+  int64_t x = 0, sign = 1;
+  for (size_t i = 1; i < n - 1; i++) {
+    if (s[i] == '-') {
+      sign *= -1;
+    }
+    else if ('0' <= s[i] && s[i] <= '9') {
+      x = x * 10 + (s[i] - '0');
+    }
+    else if (s[i] == ',') {
+      args.emplace_back(sign * x);
+      sign = 1;
+      x = 0;
+    }
+    else {
+      AT_ERROR("Illegal char in IntList default value: ", s[i]);
+    }
+  }
+  args.emplace_back(sign * x);
```

Comment:
This also seems broken when `s = "{}"`

Reply:
yeah, I agree, I probably will return an empty vector in this case. But for some reasons s = "{}" doesn't go through this parser... I haven't figured out why.

Reply:
also, other invalid default args (e.g.,`"1,2}", "{1,2-}"`) also not going through this parser..

Reply:
IIRC, we handle default value of `"{}"` as a special case in codegen.

Reply:
Did you figure out why it doesn't go through the parser?  Is it because there's no "{}" in the python_arg_parser specification?

Reply:
no, I haven't.. But will take a closer look on that